source/common/http/async_client_impl.h

```diff
@@ -300,6 +303,7 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     const std::string& cluster_name_;
     absl::optional<std::chrono::milliseconds> timeout_;
     static const absl::optional<ConnectConfig> connect_config_nullopt_;
+    static const absl::optional<Router::RouteStatsConfig> route_stats_config_nullopt_;
```

Comment:
I am not sure all the usages of absl::optional are doing what we expect. See https://abseil.io/tips/123 -- this will actually consume all the raw storage for the RouteStatsConfig even if its left as nullopt.

I realize that this is common practice including some code I've written, but it might be that in this case (and others) std::unique_ptr would be a better choice if memory conservation is the goal.

Reply:
This was one of my questions earlier https://github.com/envoyproxy/envoy/pull/21302#pullrequestreview-984949705. @mattklein123 felt it was fine. But I am OK to change if you think that helps.